src/wp-includes/block-supports/layout.php

```diff
@@ -92,7 +154,17 @@ function wp_render_layout_support_flag( $block_content, $block ) {
 		1
 	);
 
-	return $content . '<style>' . $style . '</style>';
+	// Ideally styles should be loaded in the head, but blocks may be parsed
+	// after that, so loading in the footer for now.
+	// See https://core.trac.wordpress.org/ticket/53494.
+	add_action(
+		'wp_footer',
+		function () use ( $style ) {
+			echo '<style>' . $style . '</style>';
+		}
+	);
```

Comment:
Same comment and suggestions as here https://github.com/WordPress/wordpress-develop/pull/1804/files#r744150040